pkg/cli/cli.go

```diff
@@ -0,0 +1,187 @@
+	rootCmd.PersistentFlags().StringVar(&configFlags.ServiceURL, serviceURLFlag, "",
+		"Cloud orchestration service url.")
+	rootCmd.MarkPersistentFlagRequired(serviceURLFlag)
+	// Do not show a `help` command, users have always the `-h` and `--help` flags for help purpose.
+	rootCmd.SetHelpCommand(&cobra.Command{Hidden: true})
+	rootCmd.AddCommand(hostCmd)
+	return &CVDRemoteCommand{rootCmd}
+}
+
+// ExecuteNoErrOutput is a version of Execute which returns the command error instead of
+// printing it and do not exit in case of error.
+func (c *CVDRemoteCommand) ExecuteNoErrOutput() error {
+	return c.command.Execute()
+}
+
+func (c *CVDRemoteCommand) Execute() {
+	if err := c.ExecuteNoErrOutput(); err != nil {
+		c.command.PrintErrln(err)
+		os.Exit(1)
+	}
+}
+
+type opTimeoutError string
+
+func (s opTimeoutError) Error() string {
+	return fmt.Sprintf("waiting for operation %q timed out", string(s))
+}
+
+type apiCallError struct {
+	Err *apiv1.Error
+}
+
+func (e *apiCallError) Error() string {
+	return fmt.Sprintf("api call error %s: %s", e.Err.Code, e.Err.Message)
+}
+
+func newCreateHostCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates a host.",
+		RunE: func(c *cobra.Command, args []string) error {
+			return runCreateHostCommand(c)
+		},
+	}
+	return cmd
+}
+
+func runCreateHostCommand(c *cobra.Command) error {
+	serviceURL := c.InheritedFlags().Lookup(serviceURLFlag).Value.String()
+	baseURL := serviceURL + "/v1"
+	req := &apiv1.CreateHostInstanceRequest{}
+	body := &apiv1.CreateHostRequest{CreateHostInstanceRequest: req}
+	op := new(apiv1.Operation)
+	client := &http.Client{}
+	if err := doRequest(client, "POST", baseURL+"/hosts", body, op); err != nil {
+		return err
+	}
+	if err := doRequest(client, "POST", baseURL+"/operations/"+op.Name+"/wait", nil, op); err != nil {
+		return err
+	}
+	if op.Result != nil && op.Result.Error != nil {
+		err := &apiCallError{op.Result.Error}
+		return err
+	}
+	if !op.Done {
+		return opTimeoutError(op.Name)
+	}
+	ins := new(apiv1.HostInstance)
+	if err := json.Unmarshal([]byte(op.Result.Response), ins); err != nil {
+		return err
+	}
+	c.Printf("%s\n", ins.Name)
+	return nil
+}
+
+// It either populates the passed resPayload reference and returns nil error or returns an error.
+// For responses with non-2xx status code an error will be returned.
+func doRequest(
+	client *http.Client, method, url string, reqPayload interface{}, resPayload interface{}) error {
+	var body io.Reader
+	if reqPayload != nil {
+		json, err := json.Marshal(reqPayload)
+		if err != nil {
+			return err
+		}
+		body = bytes.NewBuffer(json)
+	}
+	req, err := http.NewRequest(method, url, body)
+	req.Header.Set("Content-Type", "application/json")
+	res, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	dec := json.NewDecoder(res.Body)
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errPayload := new(apiv1.Error)
```

Comment:
nit: same comment about errPayload not having to be a pointer

Reply:
kept this one as a pointer as it's returned.